cpp/src/DataStorm/NodeSessionI.cpp

```diff
@@ -87,15 +87,32 @@ namespace
 
             if (auto nodeSession = _nodeSession.lock())
             {
```

Comment:
This is the main fix in the PR.

Dispatching the `confirmCreateSession` using a different connection that the one associated with the publisher's NodeSession object is bogus. When this happened we created a new NodeSession which was unaware of the subscriber session. And them if the connection from the publisher and the relay was closed, it would not be able to inform the subscriber, which would remain in the connected state.